aws/policy/security-services.yaml

```diff
@@ -28,6 +28,8 @@ Statement:
     Condition:
       ArnLike:
         iam:PolicyArn:
+          - arn:aws:iam::aws:policy/AWSDenyAll
+          - arn:aws:iam::aws:policy/AWSXrayWriteOnlyAccess
```

Comment:
I'm unfamiliar with xray but I think I'm missing something, why are we enabling xray policies? I can't find any use of xray in any module or PR, except https://github.com/ansible-collections/community.aws/pull/63/files#diff-ea066fed329ccea6788c93b0525ff7f2R18 which I also don't see the need for.

Reply:
As far as I can tell `AWSXrayWriteOnlyAccess` is the pre-built policy needed when someone configures the 'tracing' with lambda.  Previously everything was using the role `ansible_lambda_role` which was only 'created' in one test and never deleted (and it's existence was then relied upon by 2 other tests running lambdas).  Based on what happens when I tried running with just role that is created in the lambda_policy tests I can only assume that someone manually added either a managed policy or a policy direct on the role and we relied on it never being cleaned up, and since the policy doesn't match ansible-test-* terminator never cleaned it up.

You can see this is set for a number of the lambdas:

```
  - name: test lambda config updates
    lambda:
      name: '{{lambda_function_name}}'
      runtime: nodejs10.x
      tracing_mode: Active
      handler: mini_lambda.handler
      role: ansible_lambda_role
      role: '{{ lambda_role_name }}'
    register: update_result
```

You can see the specific error message in one of the old test runs:
https://app.shippable.com/github/ansible-collections/community.aws/runs/106/9/console
```
03:11 TASK [test lambda config updates] **********************************************
03:11 task path: /root/ansible_collections/community/aws/tests/output/.tmp/integration/lambda-sTcRlb-ÅÑŚÌβŁÈ/tests/integration/targets/lambda/tasks/main.yml:126
03:11 <testhost> ESTABLISH LOCAL CONNECTION FOR USER: root
03:11 <testhost> EXEC /bin/sh -c 'echo ~root && sleep 0'
03:11 <testhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /root/.ansible/tmp `"&& mkdir /root/.ansible/tmp/ansible-tmp-1588621158.67-894-12781299014765 && echo ansible-tmp-1588621158.67-894-12781299014765="` echo /root/.ansible/tmp/ansible-tmp-1588621158.67-894-12781299014765 `" ) && sleep 0'
03:11 Using module file /root/ansible_collections/community/aws/plugins/modules/lambda.py
03:11 <testhost> PUT /root/.ansible/tmp/ansible-local-5546h2NdZ/tmpbl3kwL TO /root/.ansible/tmp/ansible-tmp-1588621158.67-894-12781299014765/AnsiballZ_lambda.py
03:11 <testhost> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1588621158.67-894-12781299014765/ /root/.ansible/tmp/ansible-tmp-1588621158.67-894-12781299014765/AnsiballZ_lambda.py && sleep 0'
03:11 <testhost> EXEC /bin/sh -c 'ANSIBLE_DEBUG_BOTOCORE_LOGS=True /tmp/python-NqgtmT-ansible/python /root/.ansible/tmp/ansible-tmp-1588621158.67-894-12781299014765/AnsiballZ_lambda.py && sleep 0'
03:12 <testhost> EXEC /bin/sh -c 'rm -f -r /root/.ansible/tmp/ansible-tmp-1588621158.67-894-12781299014765/ > /dev/null 2>&1 && sleep 0'
03:12 The full traceback is:
03:12 Traceback (most recent call last):
03:12   File "/tmp/ansible_lambda_payload_Vzeecx/ansible_lambda_payload.zip/ansible_collections/community/aws/plugins/modules/lambda.py", line 479, in main
03:12   File "/usr/local/lib/python2.7/dist-packages/botocore/client.py", line 316, in _api_call
03:12     return self._make_api_call(operation_name, kwargs)
03:12   File "/usr/local/lib/python2.7/dist-packages/botocore/client.py", line 626, in _make_api_call
03:12     raise error_class(parsed_response, operation_name)
03:12 InvalidParameterValueException: An error occurred (InvalidParameterValueException) when calling the UpdateFunctionConfiguration operation: The provided execution role does not have permissions to call PutTraceSegments on XRAY
03:12 fatal: [testhost]: FAILED! => {
03:12     "boto3_version": "1.12.11", 
03:12     "botocore_version": "1.15.11", 
03:12     "changed": false, 
03:12     "error": {
03:12         "code": "InvalidParameterValueException", 
03:12         "message": "The provided execution role does not have permissions to call PutTraceSegments on XRAY"
03:12     }, 
03:12     "invocation": {
03:12         "module_args": {
03:12             "aws_access_key": "ASIA6CCDWXDON7MDL7WJ", 
03:12             "aws_config": null, 
03:12             "aws_secret_key": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
03:12             "dead_letter_arn": null, 
03:12             "debug_botocore_endpoint_logs": true, 
03:12             "description": "", 
03:12             "ec2_url": null, 
03:12             "environment_variables": null, 
03:12             "handler": "mini_lambda.handler", 
03:12             "memory_size": 128, 
03:12             "name": "shippable-106-9", 
03:12             "profile": null, 
03:12             "region": "us-east-1", 
03:12             "role": "ansible-test-106-9-lambda", 
03:12             "runtime": "nodejs10.x", 
03:12             "s3_bucket": null, 
03:12             "s3_key": null, 
03:12             "s3_object_version": null, 
03:12             "security_token": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
03:12             "state": "present", 
03:12             "tags": null, 
03:12             "timeout": 3, 
03:12             "tracing_mode": "Active", 
03:12             "validate_certs": true, 
03:12             "vpc_security_group_ids": null, 
03:12             "vpc_subnet_ids": null, 
03:12             "zip_file": null
03:12         }
03:12     }, 
03:12     "msg": "Trying to update lambda configuration: An error occurred (InvalidParameterValueException) when calling the UpdateFunctionConfiguration operation: The provided execution role does not have permissions to call PutTraceSegments on XRAY", 
03:12     "resource_actions": [
03:12         "lambda:GetFunction", 
03:12         "sts:GetCallerIdentity", 
03:12         "lambda:UpdateFunctionConfiguration"
03:12     ], 
03:12     "response_metadata": {
03:12         "http_headers": {
03:12             "connection": "keep-alive", 
03:12             "content-length": "114", 
03:12             "content-type": "application/json", 
03:12             "date": "Mon, 04 May 2020 19:39:19 GMT", 
03:12             "x-amzn-errortype": "InvalidParameterValueException", 
03:12             "x-amzn-requestid": "98811430-b909-4c20-a56a-e3e2f4848c64"
03:12         }, 
03:12         "http_status_code": 400, 
03:12         "request_id": "98811430-b909-4c20-a56a-e3e2f4848c64", 
03:12         "retry_attempts": 0
03:12     }
03:12 }
```